### PR TITLE
Let a checkroutine sleep when "too many open files" tcp error

### PR DESCRIPTION
**Pull request by fern4lvarez**

### CODE REVIEW

**Comment by TooAngel** [comment]
Did you check if there are any update on hchecker main repository?

**Comment by fern4lvarez** [comment]
I'll check what I can get from upstream, but many parts might differ.

**Comment by fern4lvarez** [comment]
I checked, there are important stuff to add into this fork, mostly a new redis client and dependencies management, that would require a bit more of work to have added. I'd go with this changes for now, and start working on getting this updated on parallel.

File: check.go [hunk 1]
```diff
@@ -155,6 +157,12 @@ func (c *Check) PingUrl(ch chan int) {
 			newStatus = false
 			healthy = false
 			log.Println(c.FrontendKey, "Response from", c.BackendUrl, "... TCP error:", err.Error())
+			if strings.Contains(err.Error(), "too many open files") {
+				// we let the goroutine sleep if too many files are open
+				log.Println(c.FrontendKey, "Too many open files when checking", c.BackendUrl, ". gorouting sleeping.")
```
Comment (schu) [inline check.go]: Missing space after "checking"

Comment (fern4lvarez) [inline check.go]: `log.Println`, unlike `log.Printf`, sets an space between parameters.

**Comment by schu** [comment]
Why did you decide against raising the ulimit on the loadbalancers / for the process?

File: check.go (hunk 1 above)
Comment (schu) [inline check.go]: Ah, right.

**Comment by fern4lvarez** [comment]
It's mostly motivated by [this issue](https://github.com/golang/go/issues/2631) and [this comment](http://www.reddit.com/r/golang/comments/121lc9/changing_httpget_timeout_and_not_having_too_many/c6ret6b):

> I know I can change the ulimit, but that feels like cheating. What if I were making a product for distribution to users that don't have access to that command? This is a bug, as far as I am concerned. Timedout connections should be closed, and the timeout duration should be able to be set more efficiently.

**Comment by fern4lvarez** [comment]
Fixed a typo.

**Comment by TooAngel** [comment]
LGTM

**Comment by fern4lvarez** [comment]
Feel free to merge.

**Pull request merged**